src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -193,10 +193,16 @@ private UncommonFields CreateUncommonFields()
                 retVal._lazyObsoleteAttributeData = ObsoleteAttributeData.Uninitialized;
             }
 
-            if (!_packedFlags.IsUseSiteDiagnosticPopulated)
-            {
-                retVal._lazyUseSiteDiagnostic = CSDiagnosticInfo.EmptyErrorInfo; // Indicates unknown state.
-            }
+            //
+            // Do not set _lazyUseSiteDiagnostic !!!!
+            //
+            // "null" Indicates "no errors" or "unknown state".
+            // and we know which one of the states we have from IsUseSiteDiagnosticPopulated
+            //
+            // setting _lazyUseSiteDiagnostic to a sentinel value here would introduce
+            // a number of states with various permutations between IsUseSiteDiagnosticPopulated, UncommonFields and _lazyUseSiteDiagnostic
```

Comment:
I don't see a race condition here. The constructed value `retVal` isn't exposed to any other thread until its initialization is complete. Where is the problem that is being fixed?